src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -521,12 +521,21 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
             }
             else if (SyntaxFacts.IsDeclarationExpressionType(expression, out DeclarationExpressionSyntax parent))
             {
-                if (parent.Designation.Kind() != SyntaxKind.SingleVariableDesignation)
+                switch (parent.Designation.Kind())
                 {
-                    return SymbolInfo.None;
-                }
+                    case SyntaxKind.SingleVariableDesignation:
+                        return GetSymbolInfoFromSymbolOrNone(TypeFromVariable((SingleVariableDesignationSyntax)parent.Designation, cancellationToken));
+
+                    case SyntaxKind.DiscardDesignation:
+                        return GetSymbolInfoFromSymbolOrNone(GetTypeInfoWorker(parent, cancellationToken).Type);
 
-                return TypeFromVariable((SingleVariableDesignationSyntax)parent.Designation, cancellationToken);
+                    case SyntaxKind.ParenthesizedVariableDesignation:
+                        if (((TypeSyntax)expression).IsVar)
+                        {
+                            return SymbolInfo.None;
```

Comment:
This special case is added to preserve current behavior, which appears to be intentional. See tests VarAliasInVarDeconstructionInScript and VarTypeInVarDeconstructionInScript in CodeGenDeconstructTests.cs.